### PR TITLE
refactor: cleanup BeadSort

### PR DESCRIPTION
**Pull request by alxkm**
Cleanup BeadSort and BeadSortTest, adding javadocs.

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

File: src/main/java/com/thealgorithms/sorts/BeadSort.java [hunk 1]
```diff
@@ -1,40 +1,53 @@
...
-// BeadSort can't sort negative number, Character, String. It can sort positive number only
+import java.util.Arrays;
 
+/**
+ * BeadSort cannot sort negative numbers, characters, or strings.
+ * It only works for non-negative integers.
+ *
+ * @see <a href="https://en.wikipedia.org/wiki/Bead_sort">BeadSort Algorithm (Wikipedia)</a>
+ */
 public class BeadSort {
```
Comment (vil02) [inline src/main/java/com/thealgorithms/sorts/BeadSort.java:11]: The entire class could look something like that:
```java
public class BeadSort {
    /**
     * Sorts the given array using the BeadSort algorithm.
     *
     * @param array The array of non-negative integers to be sorted.
     * @return The sorted array.
     * @throws IllegalArgumentException If the array contains negative numbers.
     */
    public int[] sort(int[] array) {
        allInputsMustBeNonnegative(array);
        return extractSortedFromGrid(fillGrid(array));
    }

    private void allInputsMustBeNonnegative(final int[] array) {
        if (Arrays.stream(array).anyMatch(s -> s < 0)) {
            throw new IllegalArgumentException("BeadSort cannot sort negative numbers.");
        }
    }

    private char[][] fillGrid(final int[] array) {
        final var maxValue = Arrays.stream(array).max().orElse(0);
        var grid = getEmptyGrid(array.length, maxValue);

        int[] count = new int[maxValue];
        for (int i = 0, arrayLength = array.length; i < arrayLength; i++) {
            int k = 0;
            for (int j = 0; j < array[i]; j++) {
                grid[count[maxValue - k - 1]++][k] = '*';
                k++;
            }
        }
        return grid;
    }

    private char[][] getEmptyGrid(final int arrayLength, final int maxValue) {
        char[][] grid = new char[arrayLength][maxValue];
        for (int i = 0; i < arrayLength; i++) {
            for (int j = 0; j < maxValue; j++) {
                grid[i][j] = '-';
            }
        }

        return grid;
    }

    private int[] extractSortedFromGrid(final char[][] grid) {
        int[] sorted = new int[grid.length];
        for (int i = 0; i < grid.length; i++) {
            int k = 0;
            for (int j = 0; j < grid[grid.length - 1 - i].length && grid[grid.length - 1 - i][j] == '*'; j++) {
                k++;
            }
            sorted[i] = k;
        }
        return sorted;
    }
}
```

Please consider the datatype of `grid`: maybe it should be an array of `boolean` or some enum (enum should be more readable).

File: src/main/java/com/thealgorithms/sorts/BeadSort.java [hunk 2]
```diff
@@ -1,40 +1,53 @@
 package com.thealgorithms.sorts;
 
-// BeadSort Algorithm(wikipedia) : https://en.wikipedia.org/wiki/Bead_sort
-// BeadSort can't sort negative number, Character, String. It can sort positive number only
+import java.util.Arrays;
 
+/**
+ * BeadSort cannot sort negative numbers, characters, or strings.
+ * It only works for non-negative integers.
```
Comment (vil02) [inline src/main/java/com/thealgorithms/sorts/BeadSort.java:7]:
```suggestion
 * BeadSort works for non-negative integers only.
```

File: src/test/java/com/thealgorithms/sorts/BeadSortTest.java [hunk 3]
```diff
@@ -1,42 +1,31 @@
...
+import static org.junit.jupiter.api.Assertions.assertThrows;
 
+import java.util.stream.Stream;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
 
 public class BeadSortTest {
     // BeadSort can't sort negative number, Character, String. It can sort positive number only
```
Comment (vil02) [inline src/test/java/com/thealgorithms/sorts/BeadSortTest.java:13]:
```suggestion
```

File: src/test/java/com/thealgorithms/sorts/BeadSortTest.java [hunk 4]
```diff
@@ -1,42 +1,31 @@
...
     }
 
     @Test
-    public void bogoSortDuplicateIntegerArray() {
-        int[] inputArray = {6, 1, 27, 15, 23, 27, 36, 23};
-        int[] outputArray = beadSort.sort(inputArray);
-        int[] expectedOutput = {1, 6, 15, 23, 23, 27, 27, 36};
-        assertArrayEquals(outputArray, expectedOutput);
+    public void testWithNegativeNumbers() {
+        assertThrows(IllegalArgumentException.class, () -> new BeadSort().sort(new int[] {3, -1, 4, 1, 5, -9}));
```
Comment (vil02) [inline src/test/java/com/thealgorithms/sorts/BeadSortTest.java:29]: Single negative number is _harder to find_.

```suggestion
        assertThrows(IllegalArgumentException.class, () -> new BeadSort().sort(new int[] {3, 1, 4, 1, 5, -9}));
```

**Review by vil02 — CHANGES_REQUESTED** [review]

File: src/main/java/com/thealgorithms/sorts/BeadSort.java (hunk 2 above)
Comment (alxkm) [reply src/main/java/com/thealgorithms/sorts/BeadSort.java]: Looks like this is already included into main description for sort method in javadocs

File: src/main/java/com/thealgorithms/sorts/BeadSort.java (hunk 1 above)
Comment (alxkm) [reply src/main/java/com/thealgorithms/sorts/BeadSort.java:5]: Thanks. Fixed this using enum. It really more readable.

**Review by vil02 — APPROVED** [review]

**Pull request merged**